L.Control.Geonames.js

```diff
@@ -54,7 +56,8 @@ L.Control.Geonames = L.Control.extend({
                     }
                 }
                 else {
-                    this._close();
+                    L.DomUtil.removeClass(this._resultsList, 'hasResults'); //close the results list if they click elsewhere
```

Comment:
This doesn't appear to be working as expected.  Clicking elsewhere does not close the results list.  The link is not the thing to attach to - we should probably add a click handler on the map instead, so that clicking on the map closes the menu.  I'll show you where to add this.